src/bioext/elastic_utils.py

```diff
@@ -11,25 +11,38 @@ def __init__(self, *args, **kwargs) -> None:
         self.session.proxies = {"http": self.proxy_endpoint, "https":self.proxy_endpoint}
 
 class ElasticsearchSession:
-    def __init__(self, server=None):
+    def __init__(self, proxy=None, conn_mode:str="HTTP"):
         requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
 
-        self.api_id = os.getenv("ELASTIC_API_ID")
-        self.api_key = os.getenv("ELASTIC_API_KEY")
-        self.es_server = server or os.getenv("ELASTIC_SERVER", "https://sv-pr-elastic01:9200") # set to GSTT server by default
+        self.es_server = os.getenv("ELASTIC_SERVER", "https://sv-pr-elastic01:9200") # set to GSTT server by default
 
-        self.proxy_node = GsttProxyNode
+        if proxy:
+            self.proxy_node = GsttProxyNode
+        else:
+            self.proxy_node = None
 
-        self.es = self.create_session()
+        if conn_mode == "API":
+            self.api_id = os.getenv("ELASTIC_API_ID")
+            self.api_key = os.getenv("ELASTIC_API_KEY")
```

Comment:
Adding some error handling

```suggestion
        try:
            if conn_mode == "API":
                self.api_id = os.getenv("ELASTIC_API_ID")
                self.api_key = os.getenv("ELASTIC_API_KEY")
                
                if not all([self.api_id, self.api_key]):
                    raise ValueError("Check that ELASTIC_API_ID and ELASTIC_API_KEY are in env variables")
```

Reply:
Could also just use `os.environ[]` which will KeyError if it doesn't exist